### ISSUE TITLE
Add an N-Gram tokenizer in Quickwit

### ISSUE THREAD

**Issue by fulmicoton** [quickwit-oss/quickwit#3392]
Add an n-gram toknizer within quickwit. 

It is already present in tantivy.
We probably can have something that goes

We probably do not need to document its presence at the moment.

This tokenizer has parameters.
(min_ngram max_ngram and prefix only)

@drusellers which parameters would you want here?

**Comment by drusellers** [comment]
This is what I'm thinking:

A tri-gram configuration

```yaml
# elided

doc_mapping:
  field_mappings:
    - name: name
      type: text
      tokenizer: n_gram
      min_ngram: 3
      max_ngram: 3
      prefix_only: false
```

Then for an edge gram you could do


```yaml
# elided

doc_mapping:
  field_mappings:
    - name: name
      type: text
      tokenizer: n_gram
      min_ngram: 3
      max_ngram: 10
      prefix_only: true
```

**Comment by fulmicoton** [comment]
Can we just add a ngramntokenizer with one config in the tokenizer manager?

**Comment by evanxg852000** [comment]
Sure what config do you have in mind (3,3, false)?

**Comment by fmassot** [comment]
you need to ask that to @drusellers...

**Comment by drusellers** [comment]
@fmassot A+

**Comment by fmassot** [comment]
Related PR on tantivy side that introduces a dedicated search analyzer in the text field : https://github.com/quickwit-oss/tantivy/pull/1073

**Comment by drusellers** [comment]
Yes, I do want them or'd together. If we are getting really fancy I would want more matches to mean a higher relevance, but that I don't really need, ...yet. :) I'm not sure I'm tracking on the search having fewer variations - I feel like I want the exact same variations. 

My ultimate goal is to let a user find the document that has the title `bustorganizationuserpreferencecachebyusercommandhandler` by searching with `user`. In this case a match of `use AND ser` would certainly work but so would a `use OR ser`. In general I like having the default tokenizer for a query be the same as the index time. But if I want to have a custom one at QUERY time I could set something up like in Elasticsearch where the query can specify the analyzer.

But again, the ultimate goal is to find `user` inside of `bustorganizationuserpreferencecachebyusercommandhandler` so I'll take what ever makes that happen. :)

**Comment by fmassot** [comment]
Closed by #3575
